Code/Framework/AzCore/Tests/DLLMainTest.cpp

```diff
@@ -105,9 +105,8 @@ extern "C" AZ_DLL_EXPORT void DestroyDLLTestVirtualClass()
 //////////////////////////////////////////////////////////////////////////
 
 
-extern "C" AZ_DLL_EXPORT void InitializeDynamicModule(void* azEnvironmentInstance)
+extern "C" AZ_DLL_EXPORT void InitializeDynamicModule([[maybe_unused]] void* azEnvironmentInstance)
 {
-    AZ::Environment::Attach(static_cast<AZ::EnvironmentInstance>(azEnvironmentInstance));
     AZ_Assert(AZ::Environment::GetInstance() == azEnvironmentInstance, "Invalid!");
     AZ_Printf("DLL", "InitializeDynamicModule called");
```

Comment:
I'd consider removing the `void*` parameter of `InitializeDynamicModule` altogether.  Seems like this is the only place where it's actually used.  I'd ask whether keeping this Assert gives enough value to us that we opt to keep the parameter.

Reply:
The parameter isn't unused. 
It is what is `AZ::EnironmentInstance*` that gets attached to the AZ::Environment.

Reply:
My concern here is backwards compatibility. Any custom gem that isn't using the `AZ_DECLARE_MODULE_INITIALIZATION` macro, but rather a custom definition, would break if the parameters are changed.

Reply:
In that case, they are likely calling `AZ::Environment::Attach` which would now be removed, thereby forcing an update on their end anyways.

Reply:
Fixed.